dot/state/epoch.go

```diff
@@ -17,19 +17,19 @@ import (
 )
 
 var (
-	ErrConfigNotFound     = errors.New("config data not found")
-	ErrEpochNotInMemory   = errors.New("epoch not found in memory map")
-	errHashNotInMemory    = errors.New("hash not found in memory map")
-	errEpochNotInDatabase = errors.New("epoch data not found in the database")
-	errHashNotPersisted   = errors.New("hash with next epoch not found in database")
+	ErrEpochLengthCannotBeZero     = errors.New("epoch length cannot be zero")
+	ErrConfigNotFound              = errors.New("config data not found")
+	ErrEpochNotInMemory            = errors.New("epoch not found in memory map")
+	errHashNotInMemory             = errors.New("hash not found in memory map")
+	errEpochNotInDatabase          = errors.New("epoch data not found in the database")
+	errHashNotPersisted            = errors.New("hash with next epoch not found in database")
+	ErrNoFirstNonOriginBlock       = errors.New("no first non origin block")
+	ErrGivenTimeBeforeNetworkStart = errors.New("given time is before network start")
```

Comment:
I would only really add exportable errors to the package signature if they are useful for outside packages to determine the type.  From checking references, seems like only `ErrConfigNotFound` is referenced outside of this package.